### ISSUE TITLE
direct_call is faster than compile

### ISSUE THREAD

**Issue by aleivag** [aleivag/pegparsing#12]
no surprise here, thats why we develop the direct call, but now knowing this, we should make compile faster

```
| tests/test_real_world/test_nodetool_status.py::test_nodetol_status_better_parser | 3.57 / 0.00             | 7.75 / 0.00             |
|  \--> direct_call                                                                |     0.78 /     0.00     |     1.26 /     0.00     |
|  \--> compile                                                                    |     2.78 /     0.00     |     6.49 /     0.00     |
| tests/test_real_world/test_nodetool_status.py::test_nodetol_status_parser        | 3.88 / 0.00             | 6.36 / 0.00             |
|  \--> direct_call                                                                |     0.78 /     0.00     |     1.39 /     0.00     |
|  \--> compile                                                                    |     3.09 /     0.00     |     4.96 /     0.00     |

**Comment by aleivag** [comment]
make direct call the only way of working in #14 , we may need to implement caching

**Comment by aleivag** [comment]
cache implemented on #15
